### ISSUE TITLE
关于boxes

### ISSUE THREAD

**Issue by letgaoyingo** [IDEA-Research/Grounded-Segment-Anything#377]
请问：boxes, logits, phrases = predict(
        model=model,
        image=image,
        caption=text_prompt,
        box_threshold=BOX_THRESHOLD,
        text_threshold=TEXT_THRESHOLD,
        device=DEVICE,
    )
中boxes返回结果为类似：[0.3397, 0.8803, 0.0617, 0.1386]，这些数值单位是什么，与锚框box在图片中的像素怎么转换？

**Comment by rentainhe** [comment]
> 请问：boxes, logits, phrases = predict( model=model, image=image, caption=text_prompt, box_threshold=BOX_THRESHOLD, text_threshold=TEXT_THRESHOLD, device=DEVICE, ) 中boxes返回结果为类似：[0.3397, 0.8803, 0.0617, 0.1386]，这些数值单位是什么，与锚框box在图片中的像素怎么转换？

是预测框相对于图像的坐标，应该是 X Y W H, 以左上角为起始点的相对坐标

**Comment by letgaoyingo** [comment]
> > 请问：boxes, logits,phrases = Predict( model=model, image=image, title=text_prompt, box_threshold=BOX_THRESHOLD, text_threshold=TEXT_THRESHOLD, device=DEVICE, ) 中boxes返回结果为类似：[0.3397, 0.8803, 0.0617, 0.1386 ]，这些数值单位是什么，与某个方框在图片中的像素怎么转换？
> 
> 是框预测相对于图像的坐标，应该是XYWH，以左上角为起始点的相对坐标

感谢回复！我理解也应该是这样的，但是图片的坐标为像素值，数值都为整型，但是boxes的返回结果为小数，例如[0.3397, 0.8803, 0.0617, 0.1386 []。原本以为这是以图片长和宽均为1的比例值，但是，经测试一个图片及锚框，这个好像比例对应不上。

例如，以下图片为锚框tensor([[0.3397, 0.8803, 0.0617, 0.1386]]) tensor([0.8276]) ['bird']及图片URL：
http://zrbhq.net:22080/data/db_lb/sm/record/12195/6f6a80ea060b59cdcd5965e9c89deeec8f7b7258.jpg

显然，图中锚框的xMin,xMax,yMin,yMax显然不是[0.3397, 0.8803, 0.0617, 0.1386]。

**Comment by letgaoyingo** [comment]
> 这是yolo的格式，cx,cy,w,h 对应图像长宽做完归一化的结果，想整回xMin,xMax,yMin,yMax如下 box = np.array([[0.3397, 0.8803, 0.0617, 0.1386]]) h,w = img.shape[:2] box[:,:2] -= box[:,2:] / 2 box[:,2:] += box[:,:2] box[:,[0,2]] *= w box[:,[1,3]] *= h box = box.astype("int")

非常感谢！

**Comment by rentainhe** [comment]
感谢矫正~
